### PR TITLE
fix lots of 'Structure does not match' warnings in ci

### LINKED ISSUE #55500: Many `Structure does not match` in `stateless_tests_flaky_check (asan)`
right now we have lots of `Warning` log `Structure does not match` in the `stateless_tests_flaky_check` . For example https://s3.amazonaws.com/clickhouse-test-reports/55467/bb204ed0bd08203621d12fe92c17e9385b788a6a/stateless_tests_flaky_check__asan_/run.log , the ratio right now is 0.434%, just lower than 1% in our trash log detect test `00002_log_and_exception_messages_formatting`

Why do we have many Structure does not match ? I checked the log:
```
2023.10.11 04:12:15.525576 [ 914 ] {} <Warning> system.asynchronous_metric_log_sender.DirectoryMonitor.default: Structure does not match (remote: pull_request_number UInt32 UInt32(size = 0), commit_sha String String(size = 0), check_start_time DateTime UInt32(size = 0), check_name LowCardinality(String) ColumnLowCardinality(size = 0, UInt8(size = 0), ColumnUnique(size = 1, String(size = 1))), instance_type LowCardinality(String) ColumnLowCardinality(size = 0, UInt8(size = 0), ColumnUnique(size = 1, String(size = 1))), instance_id String String(size = 0), event_date Date UInt16(size = 0), event_time DateTime UInt32(size = 0), metric LowCardinality(String) ColumnLowCardinality(size = 0, UInt8(size = 0), ColumnUnique(size = 1, String(size = 1))), value Float64 Float64(size = 0), local: pull_request_number UInt16 UInt16(size = 0), commit_sha String String(size = 0), check_start_time DateTime('UTC') UInt32(size = 0), check_name String String(size = 0), instance_type String String(size = 0), instance_id String String(size = 0), event_date Date UInt16(size = 0), event_time DateTime UInt32(size = 0), metric LowCardinality(String) ColumnLowCardinality(size = 0, UInt8(size = 0), ColumnUnique(size = 1, String(size = 1))), value Float64 Float64(size = 0)), implicit conversion will be done
```
We create distributed table `asynchronous_metric_log_sender` locally and sync it to remote side. But the local string column `commit_sha` (and other string columns) are of normal `String` type. which are different from the remote server with the type `ColumnLowCardinality` as shown above.

Sometimes it will cause the test of new PR flaky, if we trigger tables **xxx_sender** sync to remote side too many times. For example, in PR https://github.com/ClickHouse/ClickHouse/pull/53240

### PR DESCRIPTION
**Pull request by hanfei1991**
close #55500 

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 03d139764c7a1f7507227a40cafc68928570e810 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55503/03d139764c7a1f7507227a40cafc68928570e810/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6483541740">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55503/03d139764c7a1f7507227a40cafc68928570e810/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by alexey-milovidov — APPROVED** [review]

**Comment by hanfei1991** [comment]
update the hash suffix of tables and wait for the result

**Comment by hanfei1991** [comment]
check several tests logs. The `Structure does not match` Warning disappears.

File: docker/test/base/setup_export_logs.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ CLICKHOUSE_CI_LOGS_USER=${CLICKHOUSE_CI_LOGS_USER:-ci}
 # Pre-configured destination cluster, where to export the data
 CLICKHOUSE_CI_LOGS_CLUSTER=${CLICKHOUSE_CI_LOGS_CLUSTER:-system_logs_export}
 
-EXTRA_COLUMNS=${EXTRA_COLUMNS:-"pull_request_number UInt32, commit_sha String, check_start_time DateTime, check_name LowCardinality(String), instance_type LowCardinality(String), instance_id String, "}
+EXTRA_COLUMNS=${EXTRA_COLUMNS:-"pull_request_number UInt16, commit_sha String, check_start_time DateTime('UTC'), check_name String, instance_type String, instance_id String, "}
```
Comment (Felixoid) [inline docker/test/base/setup_export_logs.sh:18]:
```
In [16]: 1 << 16
Out[16]: 65536
```

PR numbed: #55503

pr number #50000: May 22

I feel it's *very* bad idea

Comment (hanfei1991) [reply docker/test/base/setup_export_logs.sh:18]: I will fix it ...

Comment (Felixoid) [reply docker/test/base/setup_export_logs.sh:18]: What about calculating the hash out of `EXTRA_COLUMNS` automatically too? The manual increment feels poor there

**Pull request merged**